pkg/app/var.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/kustomize/pkg/resource"
+	"github.com/kubernetes-sigs/kustomize/pkg/types"
+)
+
+func gvkn(rv types.Var) resource.ResId {
+	return resource.NewResId(rv.ObjRef.GroupVersionKind(), rv.ObjRef.Name)
+}
+
+func getFieldAsString(m map[string]interface{}, pathToField []string) (string, error) {
```

Comment:
Please make this a method on the Resource class (pkg/resource/resource.go).

Also, please rename to "GetFieldValueAsString" or simply "GetFieldValue"